src/chsh.c

```diff
@@ -129,15 +138,47 @@ static bool is_restricted_shell (const char *sh)
  */
 static bool shell_is_listed (const char *sh)
 {
-	char *cp;
 	bool found = false;
 
-#ifndef HAVE_GETUSERSHELL
-	char buf[BUFSIZ];
-	FILE *fp;
-#endif
+#if defined (USE_ECONF) && defined (VENDORDIR)
```

Comment:
Hi,

Could you edit the chsh manpage to give a brief description of the file format?